src/runtime/pkg/resourcecontrol/cgroups.go

```diff
@@ -134,8 +134,14 @@ func NewResourceController(path string, resources *specs.LinuxResources) (Resour
 	var cgroup interface{}
 	var cgroupPath string
 
+	// When the path is empty, a random string is generated
+	// as the ResourceControllerID to prevent duplicates.
+	if path == "" || path == "/" {
+		path = randStr(32)
```

Comment:
While it lowers the chance of a collision, it is still possible to get two identical paths here. I tend to agree with @fengwang666 that we need to make the function thread-safe to fix the root cause.